### ISSUE TITLE
Message Concatenation

### ISSUE THREAD

**Issue by kartman521** [nickiv/node-red-huawei-sms#4]
Hello!
I came across strange behavior of the module, and I don’t understand on what principle messages are separated and combined. For example, an SMS from an operator with information about the balance in the modem’s web interface is displayed normally and looks like one message, but for some reason this module divides these messages into 4 different ones, with the same sender number and the same date of receipt. Since the SMS is immediately deleted from the modem’s memory, they also have the same ID (40000). At the same time, the message contains a total of 209 characters, it is logical that it should be divided, but how to combine it? (Messages are in Cyrillic, for it there are not 150 characters, but only 70 maximum for one message). And the most unusual thing is that the order of the separated messages is often incorrect, instead of 1 - 2 - 3 - 4, sometimes it is 2 - 3 - 1 - 4, for example.
<img width="638" alt="Screenshot 2024-04-27 at 4 05 06 AM" src="https://github.com/nickiv/node-red-huawei-sms/assets/103675197/53f15d84-29a8-45a5-a0a8-b8d939f9a503">

**Comment by nickiv** [comment]
Thank you for the report. Frankly I did not thoroghly test long incoming messages. I will take a look what can be done.

**Comment by nickiv** [comment]
@kartman521 please check out the new version 0.0.10. You need to check "Concat long messages" in the settings of the stick. It is in config node settings (where you set password). 

If you cannot see this option after updating module, try to refresh the node-red page.

I discovered that long messages are being concatenated by lte stick itself, if it has time to do it. So I implemented a delay for such messages. 

Now long messages are not emitted by Receive node and not deleted from the stick within 30 seconds interval. In my tests this is enough for all parts of the message to arrive, and lte stick combines them just fine.

I suppose that this it the best possible approach to get long messages from hilink API. Please let me know how it works for you.

**Comment by kartman521** [comment]
Hi! After this update, not only the main problem was solved, but also the second one, which I realized only after using the module for some time, namely after turning off the modem, when 5-10 SMS could arrive at once, they were all combined into one huge message (using the join node ). Now, everything comes together perfectly, even if a lot of SMS came at once.

I am very grateful to you for this node, it works great, it has been tested on the E3372 and E3382 for the past month and a half.
